geth/jail/jail_cell.go

```diff
@@ -90,21 +89,25 @@ func (cell *JailCell) Exec(val string) (otto.Value, error) {
 
 // Run evaluates the giving js string on the associated vm llop.
 func (cell *JailCell) Run(val string) (otto.Value, error) {
+	cell.Lock()
+	defer cell.Unlock()
+
 	return cell.vm.Run(val)
 }
 
-// CellLoop returns the ottoext.Loop instance which provides underline timeout/setInternval
+// Loop returns the ottoext.Loop instance which provides underline timeout/setInternval
 // event runtime for the Jail vm.
-func (cell *JailCell) CellLoop() *loop.Loop {
+func (cell *JailCell) Loop() *loop.Loop {
+	cell.Lock()
```

Comment:
The synchronization lock being used here is a `sync.Mutex` except you wish it to be changed to a `sync.RWMutex`.